platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/junit/OtherTestExtension.java

```diff
@@ -0,0 +1,136 @@
+import org.hiero.otter.fixtures.TestEnvironment;
+import org.hiero.otter.fixtures.turtle.TurtleTestEnvironment;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.InvocationInterceptor;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.jupiter.api.extension.TestWatcher;
+
+public class OtherTestExtension implements InvocationInterceptor, ParameterResolver, TestWatcher {
+
+    /**
+     * The namespace of the extension.
+     */
+    private static final Namespace EXTENSION_NAMESPACE = Namespace.create(OtherTestExtension.class);
+
+    /**
+     * The key to store the environment in the extension context.
+     */
+    private static final String ENVIRONMENT_KEY = "environment";
+
+    /**
+     * Checks if this extension supports parameter resolution for the given parameter context.
+     *
+     * @param parameterContext the context of the parameter to be resolved
+     * @param ignored the extension context of the test (ignored)
+     *
+     * @return true if parameter resolution is supported, false otherwise
+     *
+     * @throws ParameterResolutionException if an error occurs during parameter resolution
+     */
+    @Override
+    public boolean supportsParameter(
+            @NonNull final ParameterContext parameterContext, @Nullable final ExtensionContext ignored)
+            throws ParameterResolutionException {
+        Objects.requireNonNull(parameterContext, "parameterContext must not be null");
+
+        return Optional.of(parameterContext)
+                .map(ParameterContext::getParameter)
+                .map(Parameter::getType)
+                .filter(TestEnvironment.class::isAssignableFrom)
+                .isPresent();
+    }
+
+    /**
+     * Resolves the parameter of a test method, providing a {@link TestEnvironment} instance when needed.
+     *
+     * @param parameterContext the context of the parameter to be resolved
+     * @param extensionContext the extension context of the test
+     *
+     * @return the resolved parameter value
+     *
+     * @throws ParameterResolutionException if an error occurs during parameter resolution
+     */
+    @Override
+    public Object resolveParameter(
+            @NonNull final ParameterContext parameterContext, @NonNull final ExtensionContext extensionContext)
+            throws ParameterResolutionException {
+        Objects.requireNonNull(parameterContext, "parameterContext must not be null");
+        Objects.requireNonNull(extensionContext, "extensionContext must not be null");
+
+        return Optional.of(parameterContext)
+                .map(ParameterContext::getParameter)
+                .map(Parameter::getType)
+                .filter(t -> t.equals(TestEnvironment.class))
```

Comment:
Just out of curiosity. Why are we using `Class.isAssignableFrom()` in `supportParameter()` but an equality check here?

Reply:
Good catch, added equals for both